osu.Framework/Graphics/UserInterface/TextBox.cs

```diff
@@ -557,18 +559,18 @@ private void insertString(string value, Action<Drawable> drawableCreationParamet
         protected abstract void NotifyInputError();
 
         /// <summary>
-        /// Invoked whenever a text string has been inserted to <see cref="Text"/>.
+        /// Invoked whenever pending text input from user has been consumed, this is not necessarily what has been inserted to <see cref="Text"/> (i.e. some may have been disallowed by <see cref="CanAddCharacter"/>)
```

Comment:
```suggestion
        /// Invoked when new text is added via user input```

Rename event to `OnUserTextAdded`. no need to call it consumed.